extensions/android-paging3/src/main/java/com/squareup/sqldelight/android/paging3/OffsetQueryPagingSource.kt

```diff
@@ -37,7 +37,7 @@ internal class OffsetQueryPagingSource<RowType : Any>(
       val key = params.key ?: 0L
       transacter.transactionWithResult {
         val count = countQuery.executeAsOne()
-        if (key >= count) throw IndexOutOfBoundsException()
```

Comment:
I have a filtering applied (show only unread items). When I mark an item as read obviously `count` is smaller than `key` and I'd expect that item to disappear. Paging2 source handled this perfectly well, what's the reason for throwing here @kevincianfarini ? Does paging3 not support items being removed?

Reply:
How are you making the item as read? If you're mutating a row in the database, I would expect this paging source to get invalidated.

Reply:
Yes I am mutating a row in the database. I found a way to make this work in https://github.com/cashapp/sqldelight/pull/2913 for me locally (there's some more information)